ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -876,3 +881,31 @@ func (ipfs *Connector) SwarmPeers() (api.SwarmPeers, error) {
 	}
 	return swarm, nil
 }
+
+// BlockPut triggers an ipfs block put on the given data, inserting the block
+// into the ipfs daemon's repo.
+func (ipfs *Connector) BlockPut(data []byte) (api.Pin, error) {
+	pin := api.Pin{}
+
+	r := ioutil.NopCloser(bytes.NewReader(data))
```

Comment:
It is not obvious to me why a NopCloser is used here?

Reply:
NewReaderFile takes in io.ReadCloser https://github.com/ipfs/go-ipfs-cmdkit/blob/master/files/readerfile.go#L20

Reply:
Got it now, thanks